### DISCUSSION TITLE
Should "aud" claim be required for Access JWT?

### DISCUSSION THREAD

**Discussion by maximivanov** [panva/jose#75] [Q&A]
If I'm not missing anything, currently JWT access token verification will fail if it doesn't have `aud` claim:
 https://github.com/panva/jose/blob/75e331d28a104713787381750bcb47662c2b49f2/lib/jwt/verify.js#L110

This is the case with access tokens issued by AWS Cognito. They have `client_id` but not `aud`. 

I found this SO question https://stackoverflow.com/questions/53148711/why-doesnt-amazon-cognito-return-an-audience-field-in-its-access-tokens. One of the answers raises a point that `aud` doesn't seem to be required by specification.

Looking at https://tools.ietf.org/html/rfc7519#section-4.1.3, the verbiage is:

> If the principal processing the claim does not identify itself with a value in the "aud" claim when this claim is present, then the JWT MUST be rejected.
...
Use of this claim is OPTIONAL.

What's your take on this?

**Comment by panva** [comment]
This is the profile that’s referenced by the readme to be validated with the jwt+AT profile

https://tools.ietf.org/html/draft-ietf-oauth-access-token-jwt

Other proprietary implementations should not be using the profile option and be validated as a generic JWT with options as demanded by the producer
